File: Math/9-PalinedromeNumber.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        temp = x
        rev = 0
        last = 0
        print(rev)
        while temp > 0:
            last = temp % 10
            rev = (rev*10)+last
            temp = temp//10

        if rev == x:
            return True
        else:
            return False


f = Solution()
print(f.isPalindrome(121))

File: LeetCode/Math/9-PalinedromeNumber.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    bool isPalindrome(int x)
    {
        if (x < 0)
        {
            return false;
        }
        long num = x;
        long rev = 0;

        while (num > 0)
        {
            long last = num % 10;

            rev = (rev * 10) + last;

            num = num / 10;
        }

        return (x == rev) ? true : false;
    }
};

int main()
{

    Solution s;
    bool con = s.isPalindrome(121);
    if (con == 0)
    {
        cout << "not pal";
    }
    else
    {

        cout << "pal";
    }

    return 0;
}